src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,118 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin("Ivan_Ivanov");
+        user.setAge(32);
+        user.setPassword("Qq112345PP");
+    }
+
+    @Test
+    void register_newUser_Ok() {
+        User newUser = registrationService.register(user);
+        assertEquals(user, newUser);
+    }
+
+    @Test
+    void register_newUserSameInformation_notOk() {
+        assertEquals(user, registrationService.register(user));
+        assertThrows(RuntimeException.class, () -> registrationService.register(user));
```

Comment:
In general it's better to also check that exception messages complies with expected to avoid cases where other validation accidentally worked instead of the one that is tested